txselector/txselector.go

```diff
@@ -470,15 +470,7 @@ func (txsel *TxSelector) processL2Txs(
 					l2Txs[i].AtomicGroupID.String(),
 				))
 			}
-			// no more available slots for L2Txs, so mark this tx
-			// but also the rest of remaining txs as discarded
-			for j := i; j < len(l2Txs); j++ {
-				l2Txs[i].Info = obj.Message
-				l2Txs[i].ErrorCode = obj.Code
-				l2Txs[i].ErrorType = obj.Type
-				nonSelectedL2Txs = append(nonSelectedL2Txs, l2Txs[j])
-			}
-			break
+			continue
```

Comment:
Also in this case `break` should be better than `continue`

Reply:
Yep, the state of the transactions will be `pend` at this moment and they will continue to be `pend`, since we are not touching them, also the `nonSelectedL2TXs` is used only to update the `info` field, which is exactly what we are avoiding with this PR.

We executed all the unit tests, integration tests and also tested this sending a lot of transactions to a node, everything seems to be working as expected.

The `continue` is required to review all the transactions in the pool in order to avoid forging `atomic transaction groups` partially.